eZ/Publish/Core/settings/repository/inner.yml

```diff
@@ -202,3 +208,24 @@ services:
         arguments:
             $innerPermissionResolver: '@eZ\Publish\Core\Repository\Permission\PermissionResolver'
             $permissionCriterionResolver: '@eZ\Publish\Core\Repository\Permission\PermissionCriterionResolver'
+
+    eZ\Publish\Core\Repository\Strategy\ContentValidator\ContentValidatorStrategy:
+        arguments:
+            $contentValidators: !tagged_iterator ezplatform.spi.content.validator
+
+    eZ\Publish\Core\Repository\Validator\ContentCreateStructValidator:
+        arguments:
+            $contentMapper: '@eZ\Publish\Core\Repository\Mapper\ContentMapper'
+            $fieldTypeRegistry: '@eZ\Publish\Core\FieldType\FieldTypeRegistry'
+        tags:
+            - ezplatform.spi.content.validator
+
+    eZ\Publish\Core\Repository\Validator\ContentUpdateStructValidator:
+        arguments:
+            $contentMapper: '@eZ\Publish\Core\Repository\Mapper\ContentMapper'
+            $fieldTypeRegistry: '@eZ\Publish\Core\FieldType\FieldTypeRegistry'
+            $contentLanguageHandler: '@ezpublish.spi.persistence.language_handler'
+        tags:
+            - ezplatform.spi.content.validator
+
+    eZ\Publish\SPI\Repository\Validator\ContentValidator: '@eZ\Publish\Core\Repository\Strategy\ContentValidator\ContentValidatorStrategy'
```

Comment:
Btw, Symfony standards recomend that vendor packages never use autoconfiguration and autowiring.

Reply:
@emodric I was wondering what's the deal with that, it's occasionally a real pain when used in vendor code.

Reply:
Yeah, autowiring should just burn and die in vendor code :D

Reply:
I would not complain!

Reply:
I know autowiring can be a bit of pain if used carelessly, but as long as we use it only for direct injection and not for interface-implementation autobinding everything should be transparent.

Do you have a case where autowiring fails in our code, or at least prevents you from doing what would be possible without it?

Reply:
> The downside is that `eZ\Publish\Core\Repository\Strategy\ContentValidator\ContentValidatorStrategy` would need to implement dedicated interface for injection, so it's not gonna get tagged and injected to itself.

That is why I would like to keep it as it is. I don't think custom low level value object validators will be common. Same interface also make validators and validator strategies interchangeable.

Reply:
It's not that it fails, it's just annoying to debug most of the time. Symfony best practices recommend that it should not be used in vendor code for a reason. I have yet to see the case where it was actually helpful when diving into `vendor/`

Reply:
> It's not that it fails, it's just annoying to debug most of the time. Symfony best practices recommend that it should not be used in vendor code for a reason. I have yet to see the case where it was actually helpful when diving into `vendor/`

It's really not a black & white subject. Autowiring & autoconfiguration is a way to avoid maintenance of a long list of definitions which are redundant. As long as applied for non-ambiguous cases, everything should be fine.
In this case I was referring to an autoconfiguration which would be beneficial not just internally but for 1st and 3rd party. If autoconfiguration is set, there's no point in vendor code to define its own explicitly, as that autoconfiguration applies to vendor definitions as well.

Reply:
It is black and white. Either you follow Symfony best practices, or you don't. I have nothing against enabling autoconfiguration for 3rd party code (and by that, I mean project code residing in `src/`), but everything else just puts strain on every-day